odd_collector/adapters/duckdb/client.py

```diff
@@ -0,0 +1,86 @@
+import os
+import re
+from duckdb import connect, DuckDBPyConnection
+
+
+class DuckDBClient:
+    def __init__(self, paths: list[str]):
+        self.db_files = self.__get_db_files(paths)
+
+    def __validate_paths(self, paths: list[str]):
+        for path in paths:
+            if os.path.isfile(path):
+                yield path
+            elif os.path.isdir(path):
+                new_paths = [os.path.join(path, file) for file in os.listdir(path)]
+                yield from self.__validate_paths(new_paths)
+
+    def __get_db_files(self, paths: list[str]):
+        files = list(self.__validate_paths(paths))
+        print(files)
+        return {re.search(r"\/([^\/]*)\.", file).group(1): file for file in files}
+
+    def get_connection(self, catalog: str) -> DuckDBPyConnection:
+        return connect(self.db_files[catalog])
+
+    @staticmethod
+    def get_schemas(connection: DuckDBPyConnection, catalog: str):
+        resp = connection.sql(
+            f"SELECT schema_name FROM information_schema.schemata WHERE catalog_name = '{catalog}'"
```

Comment:
is it safety to pass arguments to sql like that? Does duckdbpy provide some escaping mechanism